### PR TITLE
[DMS-193] NuGet updates, Lint removal, and Documentation updates

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs [hunk 1]
```diff
@@ -49,9 +49,9 @@ public async Task<UpsertResult> UpsertDocument(IUpsertRequest upsertRequest)
             _logger.LogError("Unknown Error - {TraceId}", upsertRequest.TraceId);
             return new UpsertResult.UnknownFailure("Unknown Failure");
         }
-        catch
+        catch (Exception ex)
         {
-            _logger.LogError("Unknown Error - {TraceId}", upsertRequest.TraceId);
+            _logger.LogError(ex, "Unknown Error - {TraceId}", upsertRequest.TraceId);
```
Comment (stephenfuqua) [inline src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs:54]: Previously the exception was trapped and not logged. SonarLint now detects that situation.

File: src/core/EdFi.DataManagementService.Core/Middleware/CoreLoggingMiddleware.cs [hunk 2]
```diff
@@ -25,16 +25,7 @@ public async Task Execute(PipelineContext context, Func<Task> next)
...
-                JsonSerializer.Serialize(
-                    new
-                    {
-                        message = "An uncaught error has occurred",
-                        error = new { ex.Message, ex.StackTrace },
-                        traceId = context.FrontendRequest.TraceId
-                    }
-                )
-            );
+            _logger.LogError(ex, "Unknown Error - {TraceId}", context.FrontendRequest.TraceId);
```
Comment (stephenfuqua) [inline src/core/EdFi.DataManagementService.Core/Middleware/CoreLoggingMiddleware.cs:28]: Consistent with error logging in the repository.

File: src/frontend/EdFi.DataManagementService.Frontend.AspNetCore/Program.cs [hunk 3]
```diff
@@ -51,15 +55,23 @@ void InitializeDatabase(WebApplication app)
...
+                app.Logger.LogCritical(failure.Error, "Database Deploy Failure");
+                Environment.Exit(-1);
+            }
+        }
+        catch (Exception ex)
         {
-            app.Logger.LogCritical("Database Deploy Failure");
-            throw failure.Error;
+            app.Logger.LogCritical(ex, "Database Deploy Failure");
+            Environment.Exit(-1);
```
Comment (stephenfuqua) [inline src/frontend/EdFi.DataManagementService.Frontend.AspNetCore/Program.cs:74]: Previously if an error occurred - for example, password authentication failure - this threw an unhandled exception and the application hung until Control-C.

File: src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs [hunk 4]
```diff
@@ -40,18 +40,24 @@ public async Task<UpsertResult> UpsertDocument(IUpsertRequest upsertRequest)
...
                 return new UpsertResult.InsertSuccess(upsertRequest.DocumentUuid);
             }
 
-            _logger.LogError("Unknown Error - {TraceId}", upsertRequest.TraceId);
+            _logger.LogError(
+                "Upsert result count was {ResultCount} for {DocumentUuid}, cause is unknown - {TraceId}",
+                resultCount,
+                upsertRequest.DocumentUuid,
+                upsertRequest.TraceId
+            );
```
Comment (stephenfuqua) [inline src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs:55]: While looking at this class, I realized that we _do know something_ about the error, and we have an opportunity to be more clear about what happened. Also clarified some messages below this.

File: src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs [hunk 5]
```diff
@@ -62,10 +68,22 @@ public async Task<GetResult> GetDocumentById(IGetRequest getRequest)
...
+                @"SELECT edfi_doc FROM public.documents
+                WHERE document_partition_key = @document_partition_key AND document_uuid = @document_uuid;",
+                conn
+            )
+            {
+                Parameters =
+                {
+                    new("document_partition_key", PartitionKeyFor(getRequest.DocumentUuid)),
+                    new("document_uuid", getRequest.DocumentUuid.Value)
+                }
```
Comment (stephenfuqua) [inline src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs:84]: Parametrize as done with insert and update.

File: src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs [hunk 6]
```diff
@@ -40,18 +40,24 @@ public async Task<UpsertResult> UpsertDocument(IUpsertRequest upsertRequest)
...
+            _logger.LogError(
+                "Upsert result count was {ResultCount} for {DocumentUuid}, cause is unknown - {TraceId}",
+                resultCount,
+                upsertRequest.DocumentUuid,
+                upsertRequest.TraceId
+            );
             return new UpsertResult.UnknownFailure("Unknown Failure");
         }
-        catch
+        catch (Exception ex)
```
Comment (simpat-adam) [inline src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs:58]: This is no more helpful. Why are we burying the thing that could help us debug?

**Review by simpat-adam — APPROVED** [review]

**Pull request merged**